staging/src/k8s.io/apimachinery/pkg/api/validate/subfield.go

```diff
@@ -36,6 +36,11 @@ func Subfield[Tstruct any, Tfield any](ctx context.Context, op operation.Operati
 	if oldStruct != nil {
 		oldVal = getField(oldStruct)
 	}
+	// TODO: passing an equiv function to Subfield for direct comparison instead of
+	// SemanticDeepEqual if fields can be compared directly, to improve performance.
+	if op.Type == operation.Update && SemanticDeepEqual(newVal, oldVal) {
```

Comment:
There was a followup to this in dev-branch, but we can pick it up in the next sync PR

Reply:
oh, my bad!